doc/_extensions/zephyr/api_overview.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import doxmlparser
+
+from docutils import nodes
+from doxmlparser.compound import DoxCompoundKind
+from pathlib import Path
+from sphinx.application import Sphinx
+from sphinx.util.docutils import SphinxDirective
+from typing import Any, Dict
+
+
+class ApiOverview(SphinxDirective):
+    r"""
+    This is a Zephyr directive to generate a table containing an overview
+    of all APIs. This table will show the API name, version and since which
+    version it is present - all information extracted from Doxygen XML output.
+
+    It is exclusively used by the doc/develop/api/overview.rst page.
+
+    Configuration options:
+    api_overview_doxygen_xml_dir: Doxygen xml output directory
+    api_overview_doxygen_base_url: Doxygen base html directory
+    """
+
+    def run(self):
+        return [self.env.api_overview_table]
+
+
+def get_group(innergroup, all_groups):
+    return [
+        g
+        for g in all_groups
+        if g.get_compounddef()[0].get_id() == innergroup.get_refid()
+    ][0]
+
+
+def visit_group(app, group, all_groups, rows, indent=0):
+    version = since = url = ""
+    cdef = group.get_compounddef()[0]
+    ssects = [
+        s for p in cdef.get_detaileddescription().get_para() for s in p.get_simplesect()
+    ]
+    for sect in ssects:
+        if sect.get_kind() == "since":
+            since = sect.get_para()[0].get_valueOf_()
+        elif sect.get_kind() == "version":
+            version = sect.get_para()[0].get_valueOf_()
+
+    url_base = app.config.api_overview_doxygen_base_url.rstrip("/")
+    url = f"{url_base}/{cdef.get_id()}.html"
+
+    title = cdef.get_title()
+
+    row_node = nodes.row()
+
+    # Next entry will contain the spacer and the link with API name
+    entry = nodes.entry()
+    span = nodes.Text("".join(["-"] * indent) + " ")
+    entry += span
+
+    # API name with link
+    inline = nodes.inline()
+    reference = nodes.reference(text=title, refuri=url)
+    reference.attributes["internal"] = True
+    inline += reference
+    entry += inline
+    row_node += entry
+
+    # Finally, add version and since
```

Comment:
when version (... or rather status? :)) is not set, should we by default show the same value as the parent group?

Reply:
Not sure, as there's a risk of implying something is more stable than it is.